demo.py

```diff
@@ -25,115 +27,274 @@
-print("\nNew charging locations:", new_charging_nodes)
-new_cs_graph = G.subgraph(new_charging_nodes)
-
-# Display image of results for user
-#   - Black nodes: available space
-#   - Red nodes: current charger location
-#   - Nodes marked 'P': POI locations
-#   - Blue nodes: new charger locations
-
-poi_labels = {x: 'P' for x in poi_graph.nodes()}
-poi_cs_labels = {x: 'P' for x in poi_graph.nodes()}
-pos = {x: [x[0],x[1]] for x in G.nodes()}
-
-fig, (ax1, ax2) = plt.subplots(1, 2)
-fig.suptitle('New EV Charger Locations')
-
-# Draw existing set up (left image)
-nx.draw_networkx(G, ax=ax1, pos=pos, with_labels=False, node_color='k', font_color='w')
-nx.draw_networkx(poi_graph, ax=ax1, pos=pos, with_labels=True, 
-                    labels=poi_labels, node_color='k', font_color='w')
-nx.draw_networkx(cs_graph, ax=ax1, pos=pos, with_labels=False, node_color='r',
-                    font_color='k')
-nx.draw_networkx(poi_cs_graph, ax=ax1, pos=pos, with_labels=True, 
-                    labels=poi_cs_labels, node_color='r', font_color='w')
-
-# Draw new set up (right image)
-nx.draw_networkx(G, ax=ax2, pos=pos, with_labels=False, node_color='k', 
-                    font_color='w')
-nx.draw_networkx(poi_graph, ax=ax2, pos=pos, with_labels=True, 
-                    labels=poi_labels,node_color='k', font_color='w')
-nx.draw_networkx(cs_graph, ax=ax2, pos=pos, with_labels=False, node_color='r',
-                    font_color='k')
-nx.draw_networkx(poi_cs_graph, ax=ax2, pos=pos, with_labels=True, 
-                    labels=poi_cs_labels,  node_color='r', font_color='w')
-nx.draw_networkx(new_cs_graph, ax=ax2, pos=pos, with_labels=False, 
-                    node_color='#00b4d9', font_color='w')
-
-# Save image
-plt.savefig("map.png")
+def read_in_args():
+    """ Read in user specified parameters or use defaults."""
+
+    # Read in user option for random seed
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-s", "--seed", help="set a random seed for scenario", type=int)
+    parser.add_argument("-x", "--width", help="set the width of the grid", type=int)
+    parser.add_argument("-y", "--height", help="set the height of the grid", type=int)
+    parser.add_argument("-p", "--poi", help="set the number of POIs", type=int)
+    parser.add_argument("-c", "--chargers", help="set the number of existing chargers", type=int)
+    parser.add_argument("-n", "--new_chargers", help="set the number of new chargers", type=int)
+    args = parser.parse_args()
+
+    # Check all inputs are valid
+    if isinstance(args.seed, int):
+        random.seed(args.seed)
+    elif args.seed is None:
+        pass
+    else:
+        print("Seed must be an integer.")
+        sys.exit(0)
+
+    if isinstance(args.width, int) and args.width>0:
+        w = args.width
+    elif args.width is None:
+        w = 15
+    else:
+        print("Width must be a positive integer.")
+        sys.exit(0)
```

Comment:
The readout is pretty ugly for large ranges when using choices (see image), so I'll just read it in and then do the checks since I need to do some other checks anyway to make sure the grid is large enough for the other parameters specified.
<img width="562" alt="Screen Shot 2020-12-23 at 8 04 10 AM" src="https://user-images.githubusercontent.com/38987205/102969176-8b51c400-44f5-11eb-9b75-c4c7d1d98627.png">